src/payment/strategies/barclays/barclays-payment-strategy.spec.ts

```diff
@@ -0,0 +1,135 @@
+            submitOrder: jest.fn(() => of()),
+        };
+
+        paymentActionCreator = {
+            submitPayment: jest.fn(() => of()),
+        };
+
+        hostedFormFactory = {} as HostedFormFactory;
+
+        threeDSecureFlow = {
+            prepare: jest.fn(() => Promise.resolve()),
+            start: jest.fn(() => Promise.resolve()),
+        };
+
+        state = store.getState();
+
+        jest.spyOn(store, 'dispatch')
+            .mockResolvedValue(state);
+
+        jest.spyOn(store, 'getState')
+            .mockReturnValue(state);
+
+        jest.spyOn(state.paymentMethods, 'getPaymentMethodOrThrow')
+            .mockReturnValue(paymentMethod);
+
+        strategy = new BarclaysPaymentStrategy(
+            store,
+            orderActionCreator as OrderActionCreator,
+            paymentActionCreator as PaymentActionCreator,
+            hostedFormFactory,
+            threeDSecureFlow as CardinalThreeDSecureFlowV2
+        );
+    });
+
+    it('is special type of credit card strategy', () => {
+        expect(strategy)
+            .toBeInstanceOf(CreditCardPaymentStrategy);
+    });
+
+    describe('#initialize', () => {
+        it('throws error if payment method is not defined', async () => {
+            jest.spyOn(state.paymentMethods, 'getPaymentMethodOrThrow')
+                .mockImplementation(() => { throw new Error(); });
+
+            await expect(strategy.initialize({ methodId: paymentMethod.id })).rejects.toThrow(Error);
+        });
+
+        it('does not prepare 3DS flow if not enabled', async () => {
+            paymentMethod.config.is3dsEnabled = false;
+
+            await strategy.initialize({ methodId: paymentMethod.id });
+
+            expect(threeDSecureFlow.prepare)
+                .not.toHaveBeenCalled();
+        });
+
+        it('prepares 3DS flow if enabled', async () => {
+            paymentMethod.config.is3dsEnabled = true;
+
+            await strategy.initialize({ methodId: paymentMethod.id });
+
+            expect(threeDSecureFlow.prepare)
+                .toHaveBeenCalled();
+        });
+    });
+
+    describe('#execute', () => {
+        let payload: OrderRequestBody;
+
+        beforeEach(() => {
+            payload = merge({}, getOrderRequestBody(), {
+                payment: {
+                    methodId: paymentMethod.id,
+                    gatewayId: paymentMethod.gateway,
+                },
+            });
+        });
+
+        it('throws error if payment method is not defined', async () => {
+            jest.spyOn(state.paymentMethods, 'getPaymentMethodOrThrow')
+                .mockImplementation(() => { throw new Error(); });
+
+            await expect(strategy.execute(payload)).rejects.toThrow(Error);
+        });
+
+        it('does not start 3DS flow if not enabled', async () => {
+            paymentMethod.config.is3dsEnabled = false;
+
+            await strategy.execute(payload);
+
+            expect(threeDSecureFlow.start)
```

Comment:
not sure how easy it is, but would be good to assert that `parent.execute` is called?

Reply:
Do you mean the next unit: _'starts 3DS flow if enabled'_? In this case, it's not being called.

Reply:
Forget it, I got confused 😅. I added the assertion.